webapp/src/share.tsx

```diff
@@ -382,7 +384,11 @@ export class ShareEditor extends data.Component<ShareEditorProps, ShareEditorSta
                 // save project name if we've made a change change
                 p = this.props.parent.updateHeaderNameAsync(newProjectName);
             }
-            p.then(() => this.props.parent.anonymousPublishAsync(screenshotUri))
+            // if screenshots are enabled, always take one
+            if (targetTheme.simScreenshot && !screenshotUri) {
```

Comment:
we should not attach a screenshot "silently", the user should always be able to review it. This should be done before publishing.